patch-db/src/locker/mod.rs

```diff
@@ -74,15 +74,47 @@ impl Locker {
     pub async fn lock(
         &self,
         handle_id: HandleId,
-        ptr: JsonPointer,
+        ptr: JsonGlob,
         lock_type: LockType,
     ) -> Result<Guard, LockError> {
         // Pertinent Logic
-        let lock_info = LockInfo {
+        let lock_info: LockInfos = LockInfo {
             handle_id,
             ptr,
             ty: lock_type,
-        };
+        }
+        .into();
+        self._lock(lock_info).await
+    }
+
+    pub async fn lock_all(
+        &self,
+        handle_id: HandleId,
+        locks: Vec<LockTargetId>,
+    ) -> Result<Guard, LockError> {
+        // Pertinent Logic
+        let lock_info: LockInfos = LockInfos(
+            locks
+                .into_iter()
+                .zip(std::iter::repeat(handle_id))
```

Comment:
clever. I tried eliminating this because "why not just use the value you have already", then ownership kicked in, and doing this without this zip would require a move closure that cloned the captured value each time, all of this would be way less elegant than what you've done here.